src/mjolnir/valhalla_build_tiles.cc

```diff
@@ -28,115 +26,113 @@ void list_stages() {
-  }
-
-  // configure logging
-  boost::optional<boost::property_tree::ptree&> logging_subtree =
-      pt.get_child_optional("mjolnir.logging");
-  if (logging_subtree) {
-    auto logging_config =
-        valhalla::midgard::ToMap<const boost::property_tree::ptree&,
-                                 std::unordered_map<std::string, std::string>>(logging_subtree.get());
-    valhalla::midgard::logging::Configure(logging_config);
-  }
-
-  // Convert stage strings to BuildStage
-  BuildStage start_stage = string_to_buildstage(start_stage_str);
-  if (start_stage == BuildStage::kInvalid) {
-    std::cerr << "Invalid start stage" << std::endl;
-    list_stages();
-    return EXIT_FAILURE;
-  }
-  BuildStage end_stage = string_to_buildstage(end_stage_str);
-  if (end_stage == BuildStage::kInvalid) {
-    std::cerr << "Invalid end stage" << std::endl;
-    list_stages();
-    return EXIT_FAILURE;
-  }
-  LOG_INFO("Start stage = " + to_string(start_stage) + " End stage = " + to_string(end_stage));
-
-  if (input_files.size() == 0 &&
-      (start_stage <= BuildStage::kParseNodes && end_stage >= BuildStage::kParseWays)) {
-    std::cerr << "Input file is required\n\n" << options << "\n\n";
-    return EXIT_FAILURE;
-  }
-
-  // Make sure start stage < end stage
-  if (static_cast<int>(start_stage) > static_cast<int>(end_stage)) {
-    std::cerr << "Starting build stage is after ending build stage in pipeline. "
-              << " Please revise options!" << std::endl;
-    list_stages();
-    return EXIT_FAILURE;
-  }
-
-  // Build some tiles!
-  if (build_tile_set(pt, input_files, start_stage, end_stage)) {
-    return EXIT_SUCCESS;
-  } else {
-    return EXIT_FAILURE;
+    // args
+    std::string config_file_path = "";
+    std::vector<std::string> input_files;
+
+    // ref:
+    // https://github.com/jarro2783/cxxopts/blob/302302b30839505703d37fb82f536c53cf9172fa/src/example.cpp
+    cxxopts::Options options(
+        "valhalla_build_tiles " VALHALLA_VERSION,
+        "valhalla_build_tiles is a program that creates the route graph\nfrom one or multiple osm.pbf extract(s)\n");
+
+    options.add_options()("h,help",
+                          "Print this help message.")("v,version",
+                                                      "Print the version of this software.")(
+        "c,config", "Path to the configuration file",
+        cxxopts::value<std::string>())("i,inline-config", "Inline JSON config",
+                                       cxxopts::value<std::string>())(
+        "s,start", "Starting stage of the build pipeline",
+        cxxopts::value<std::string>()->default_value(
+            "initialize"))("e,end", "End stage of the build pipeline",
+                           cxxopts::value<std::string>()->default_value(
+                               "cleanup"))("f,files", "positional arguments",
+                                           cxxopts::value<std::vector<std::string>>());
+
+    options.parse_positional({"files"});
```

Comment:
since `files` is a `std::vector<std::string>>` this fetches multiple positional args